### PR TITLE
Set default weight and dimension units based on country

### LINKED ISSUE #53651: Set Default Units of Measure Based on Country
### Issue Description

p1733929400379389-slack-C01SFMVEYAK

Currently, the Products Setting page defaults the unit of measure to kilograms (kg) and centimeters (cm), regardless of the user's home country. This creates a poor user experience for users in countries like the United States, where imperial units (pounds and inches) are standard, particularly for shipping.

We should set the default unit of measure to align with the home country's standard units during the initialization process or when the country is updated via the core profiler.

### Acceptance Criteria

1. Default Units Logic: Default to imperial units for the default country (the United States).
2. Core Profiler Updates:

Core profiler updates the unit of measure automatically when the country is updated.
Logic implemented:

```
IF country = "USA":
    set_units_to_imperial()
ELSE:
    set_units_to_metric()
```

3. Clear documentation on default logic and integration with the core profiler

### PR DESCRIPTION
**Pull request by chihsuan**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/53651

We defaults to metric units (kg and cm) in the Settings > Products page, which can lead to a poor user experience for U.S.-based users who primarily use imperial units (lbs and inches).

This PR makes the following changes:

1. Ensure U.S. installations default to imperial units
2. Automatically updates the weight and dimension units in product settings when the home country is updated via the Core Profiler.
3. Corrected weight and dimension units for specific countries in locale-info.php.
   - The US primarily uses pounds (lbs) for weight and inches (in) for dimensions.
   - US territories like Guam (GU), American Samoa (AS), US Virgin Islands (VI) also use imperial units.
   - UK [officially uses metric](https://www.gov.uk/weights-measures-and-packaging-the-law), though many retailers use imperial units. 
4. Update `core-profiler.md`
5. Update tests to reflect the changes

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Review (use AI tool) the locale-info.php file for the weight and dimension units for each country and make sure the units are set correctly. 

**Test new stores setup:**

1. Create a fresh site
2. Go to `/wp-admin/admin.php?page=wc-settings&tab=products`
3. Confirm that `Weight unit` and `Dimension unit` are set to `lbs` and `in` respectively.
4. Go to Core Profiler and choose `Set up my store`
6. Update the country to `United Kingdom` on about store step.
7. Confirm that `Weight unit` and `Dimension unit` are set to `kg` and `cm` respectively.
8. Go to Core Profiler and choose `Skip guided setup`
9. Select `Guam` as the country
10. Confirm that `Weight unit` and `Dimension unit` are set to `lbs` and `in` respectively.

Optional:

- Repeat the steps above for other US territories like American Samoa (AS), US Virgin Islands (VI) and confirm that the units are set to `lbs` and `in` respectively.

**Existing stores should not be affected**

1. Create a fresh site with latest stable WooCommerce and beta tester
2. Skip or complete core profiler setup or disable core profiler
3. Check the weight and dimension units in the settings page
4. Update WooCommerce to include this PR (using live branch feature)
5. Confirm that `Weight unit` and `Dimension unit` are remain the same as before the update.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Comment by adrianduffell** [comment]
Nice work here, @chihsuan! I agree in terms of shipping, US and territories should be changed to lbs/in units, while UK should be cm/kg units. I think we will need to create a developer advisory about this in case any 3PD depend on the previous values.

File: plugins/woocommerce/client/admin/docs/features/core-profiler.md [hunk 1]
```diff
@@ -73,6 +73,16 @@ These options are set by looking up the currency data from `@woocommerce/currenc
...
+### Weight and Dimension options
+
+These options are set by looking up the weight and dimension data from admin localeInfo settings after the user has selected their country.
+
+- `woocommerce_weight_unit`
+
+- `woocommerce_dimension_unit`
+
+Refer to [class-wc-settings-products.php](https://woocommerce.github.io/code-reference/files/woocommerce-includes-admin-settings-class-wc-settings-products.html) and [locale-info.php](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/i18n/locale-info.php) for the full details of the weight and dimension settings.
+
```
Comment (adrianduffell) [inline plugins/woocommerce/client/admin/docs/features/core-profiler.md:85]: ❤️

**Review by adrianduffell — APPROVED** [review]
It tested great! Thanks @chihsuan 🚀

**Review by rjchow — APPROVED** [review]
Tested well, but I wonder if we should stage this introduction in phases - at first only change it with a parameter and post a deprecation notice if this parameter is not provided.

**Comment by chihsuan** [comment]
> but I wonder if we should stage this introduction in phases - at first only change it with a parameter and post a deprecation notice if this parameter is not provided.

Hey @rjchow Could you please explain more about this?  I'm not sure how we can change it with a parameter.  :pray:

This PR changes the following things:

1. Changing the default weight and dimension units: I don't think other plugins would need to call this function and change the default values. When WooCommerce is activated, all defaults are stored in the database so if plugins change the default values after WooCommerce is activated, it will not affect the default values.
2. Automatically updates the weight and dimension units in Core Profiler: We don't allow other plugins to change core profiler, so this is not an issue.
3. `locale-info.php` changes: This only might affect other plugins that use this file. But this is not a function. We need to include this file to use this file. I think the only way to duplicate this file and create a new version. However, I don't know if this is a good idea. I guess when other plugins use this file, they expect to use the default values as the same as Core. 

```php
	$locale_info         = include WC()->plugin_path() . '/i18n/locale-info.php';
```

I think creating a developer advisory about this is a good idea. We will also include this in the release highlight .

**Comment by rjchow** [comment]
> > but I wonder if we should stage this introduction in phases - at first only change it with a parameter and post a deprecation notice if this parameter is not provided.
> 
> Hey @rjchow Could you please explain more about this? I'm not sure how we can change it with a parameter. 🙏
> 
> This PR changes the following things:
> 
>     1. Changing the default weight and dimension units: I don't think other plugins would need to call this function and change the default values. When WooCommerce is activated, all defaults are stored in the database so if plugins change the default values after WooCommerce is activated, it will not affect the default values.
> 
>     2. Automatically updates the weight and dimension units in Core Profiler: We don't allow other plugins to change core profiler, so this is not an issue.
> 
>     3. `locale-info.php` changes: This only might affect other plugins that use this file. But this is not a function. We need to include this file to use this file. I think the only way to duplicate this file and create a new version. However, I don't know if this is a good idea. I guess when other plugins use this file, they expect to use the default values as the same as Core.
> 
> 
> ```
> 	$locale_info         = include WC()->plugin_path() . '/i18n/locale-info.php';
> ```
> 
> I think creating a developer advisory about this is a good idea. We will also include this in the release highlight .

Yea it's a good point about not being able to change the interface due to it being a directly included file.

I found two instances of usages in 3PD plugins in the all-plugins repo, but they were both currency related so it should not be an issue:

image

One way around the direct inclusion might be to set a global before the include and then check the global within `locale-info.php`. But anyway, I don't think it's likely to cause major issues so I'm ok with it as-is

**Comment by chihsuan** [comment]
> I found two instances of usages in 3PD plugins in the all-plugins repo, but they were both currency related so it should not be an issue:
> 

> One way around the direct inclusion might be to set a global before the include and then check the global within `locale-info.php`. But anyway, I don't think it's likely to cause major issues so I'm ok with it as-is

Thanks for checking! I'll go ahead and merge it.

**Pull request merged**